File: 114. Flatten Binary Tree to Linked List/main.go
package main

type TreeNode struct {
	Val   int
	Left  *TreeNode
	Right *TreeNode
}

/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */
func flatten(root *TreeNode) {
	if root == nil {
		return
	}

	flatten(root.Left)
	flatten(root.Right)

	l, r := root.Left, root.Right

	root.Left = nil
	root.Right = l

	p := root

	for p.Right != nil {
		p = p.Right
	}
	p.Right = r
}

File: 114. Flatten Binary Tree to Linked List/main.py
# Definition for a binary tree node.
class TreeNode:
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right
from typing import Optional


class Solution:
    def flatten(self, root: Optional[TreeNode]) -> None:
        """
        Do not return anything, modify root in-place instead.
        """
        def innerFlat(node: TreeNode):
            if not node:
                return
            
            if node.left:
                innerFlat(node.left)
                temp = node.right
                node.right = node.left
                curr = node.right
                while curr and curr.right:
                    curr = curr.right
                curr.right = temp
                node.left = None

            curr = node
            while curr:
                if curr.left:
                    innerFlat(curr)
                curr = curr.right
                
        innerFlat(root)

s = Solution()
s.flatten(TreeNode(1, TreeNode(2,TreeNode(3),TreeNode(4)), TreeNode(5, None, TreeNode(6))))